src/pages/components/notification/usage.mdx

```diff
@@ -4,79 +4,198 @@ description: Notifications are messages that communicate information to the user
-| _Inline notification_ | Inline notifications show up in task flows, to notify users of the status of an action. They usually appear at the top of the primary content area.                   |
+#### Sizing
 
-### High vs. low contrast
+The width of inline notifications varies based on content and layout. They can expand to the fill the container or content area they relate to. Their height is based on the content length, which should not exceed two lines of text.
 
-Use high-contrast style notifications for critical messaging. Low-contrast notifications are best used for supplemental messaging and are less visually disruptive to users. When in doubt, use the low contrast style.
+#### Placement
 
-Inline and toast notifications can use different styles but you should never mix styles within the variations.
+Inline notifications appear near their related items. They can expand to fill the width of the container or content area they are in and should align to the grid columns.
+
+We recommend placing inline notifications at the bottom of forms, just above the submission and cancel buttons. When error notifications apply to individual text inputs, they should supplement the error state on that specific input field.
 
 <Row>
 <Column colLg={8}>
 
-![High vs. low contrast](images/notification-usage-4.png)
+![Example of inline notification placement](images/notification-placement-inline.png)
+
+<Caption>
+  Example of an inline notification supporting an error message on a form
+</Caption>
 
 </Column>
 </Row>
 
-## Placement
+### Behavior
 
-#### Toast notifications
+#### Dismissal
 
-Toast notifications slide in and out a page from the top-right corner. Actionable notifications do not appear on mobile screen widths.
+Inline notifications do not dismiss automatically. They persist on the page until the user dismisses them or takes action that resolves the notification.
+
+A small "x" in the top right corner is used to dismiss inline notifications. Including the close button is optional and should not be included if it is critical for a user to read or interact with the notification.
+
+## Toast notifications
+
+### Formatting
 
 <Row>
 <Column colLg={8}>
 
-![Toast notification example](images/notification-usage-2.png)
+![Example of toast notification with action button and time stamp](images/notification-toast-action.png)
 
 </Column>
 </Row>
 
-#### Inline notifications
+#### Action
+
+Toast notifications can include a link at the end of their body content. This link should be short and navigate users to a page or modal where they can take action to address the notification or find further information. Because toast notifications automatically dismiss, there should be alternative routes to navigate to the link destination.
+
+If you include links within the notification body that redirect the user to steps, be sure keyboard and screenreader users have enough time to access the notification or can easily find the notification after it disappears.
```

Comment:
In general, it seems like we would want to recommend that notifications don't have any interactive content within them. When using a live region, the text of the area is what is communicated to the end user:

![demo](https://user-images.githubusercontent.com/3901764/78923406-c7272500-7a5d-11ea-9398-24083ab5dd95.gif)

As a result, there is no way for them to know if the content is interactive or not nor is there an easy way for them to navigate to it.

Reply:
More specifically. Is the feedback here that this guidance should change because we don't/can't include links in notifications?

Reply:
This is technically speaking something that we currently allow and show in the storybook. She is only commenting on what we're currently allowing. If we don't want to allow this then we need to also remove this from the component code. 

![image](https://user-images.githubusercontent.com/11670886/78930664-1bd09d00-7a6a-11ea-9308-cc78a6a66267.png)

Reply:
@aagonzales definitely agreed, I think this needs to be a conversation for us all to better understand what is/is not applicable here. It feels odd given that so many sites use this pattern that this is how a screen reader behaves, maybe I'm misunderstanding something? Was surprised when I was testing it out in that recording above that it announced the way that it did.

@dakahn do you have any insights here? Clearly this is a common pattern, including notifications at the OS-level, is there anything we can do to accommodate interactive content either through a link/action button/close button?

Reply:
**At IBM we have the requirement that any mouse interaction have a keyboard equivalent.** With notifications that popup and disappear (toasts or whatever) and have interactable child elements this rule becomes problematic. 

How do we move keyboard focus to that notification? If this were a modal dialog we'd simply steal the users focus and _move it to the dialog_ when the modal was opened and then trap focus in the dialog until an action was taken. Toasts come and go as the user works and are intended to be unobtrusive so the above described interaction pattern doesn't really work. 

At the application level there are ways to solve around this problem (having notifications collected somewhere that _is keyboard accessible_ like this:

![image](https://user-images.githubusercontent.com/40970507/78946424-4af60700-7a88-11ea-98bc-2810eb0c7435.png)

but its always highly inadvisable to use operating system level interactions as guidance for building websites since -- in short -- they are very different contexts and only superficially comparable.

Reply:
@dakahn it seems like the guidance from WAI-ARIA gets distilled into two categories: alerts and alert dialogs:

> An alert is an element that displays a brief, important message in a way that attracts the user's attention without interrupting the user's task
> Because alerts are intended to provide important and potentially time-sensitive information without interfering with the user's ability to continue working, it is crucial they do not affect keyboard focus.

> The alert dialog is designed for situations where interrupting work flow is necessary.

---

Alerts are what we can use for time-based notifications, but if there is anything interactive then we need to use the alert dialog?

What happens if we want to use a timed alert, but also want to give someone the ability to dismiss it early? This seems like it'd directly contradict the alert guidance as it says that it shouldn't steal focus from the user.

**Links & resources**

- [Alert example](https://www.w3.org/TR/wai-aria-practices/examples/alert/alert.html)
- [Alert dialog example](https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/alertdialog.html)

Reply:
Most common example I can think of is the service work content refresh confirmation or a cookie approval form.

A notification tray isn’t enough for these types of product-level notifications/approvals. A modal on the other hand is far too invasive. I don’t see the problem with adding actions to our Toast notifications so long as we prevent their automatic dismissal.

Reply:
> What happens if we want to use a timed alert, but also want to give someone the ability to dismiss it early?

They're highly problematic for sure. You can minimize a lot of these gnarly UX frustrations for keyboard/screen reader/magnification users, but some of those solutions cause frustrations for other (dialogs with interactable elements becoming modal for example solves one problem but creates another).